hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -63,7 +63,8 @@ public ListingBasedRollbackStrategy(HoodieTable table,
             table.getMetaClient().getBasePath(), config);
       } else {
         rollbackRequests = RollbackUtils
-            .generateRollbackRequestsUsingFileListingMOR(instantToRollback, table, context);
+            .generateRollbackRequestsUsingFileListingMOR(
```

Comment:
may I know is there any particular reason to pass slice view from here? why not pass in table and let generateAppendRollbackBlocksAction get the slice view from table?

Reply:
there are too many arguments in there. Can we see if we can reduce them.

Reply:
The reason to refactor the rollback methods was to avoid upgrade and downgrade helpers in the engine client package.  As discussed offline, I keep one helper class for engine-specific logic needed for the upgrade and downgrade actions.  Based on that I keep these rollback methods the same now.